0.9.2: test_get_body_barycentric fails -- 0. != 360.

Hi,

I now managed to upload the latest version to Debian, and found the following test failure:
```
__________________________ test_get_body_barycentric ___________________________

    def test_get_body_barycentric():
        # Validate against published values from the Astronomical Almanac (2013)
        e1 = get_body_heliographic_stonyhurst('earth', '2013-Jan-01')
        assert_quantity_allclose(e1.lon, 0*u.deg, atol=1e-12*u.deg)
        assert_quantity_allclose(e1.lat, -3.03*u.deg, atol=5e-3*u.deg)
        assert_quantity_allclose(e1.radius, 0.9832947*u.AU, atol=5e-7*u.AU)
    
        e2 = get_body_heliographic_stonyhurst('earth', '2013-Sep-01')
>       assert_quantity_allclose(e2.lon, 0*u.deg, atol=1e-12*u.deg)

sunpy/coordinates/tests/test_ephemeris.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

actual = <Longitude 360. deg>, desired = <Quantity 0. deg>, rtol = 1e-07
atol = <Quantity 1.e-12 deg>, kwargs = {}
np = <module 'numpy' from '/usr/lib/python2.7/dist-packages/numpy/__init__.pyc'>

    def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=None,
                                 **kwargs):
        """[...]"""
        import numpy as np
        np.testing.assert_allclose(*_unquantify_allclose_arguments(actual, desired,
                                                                   rtol, atol),
>                                  **kwargs)
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=1e-12
E       
E       (mismatch 100.0%)
E        x: array(360.)
E        y: array(0.)

/usr/lib/python2.7/dist-packages/astropy/tests/helper.py:510: AssertionError
```

This is on a 32 bit x86 platform. Full build log [here](https://buildd.debian.org/status/fetch.php?pkg=sunpy&arch=i386&ver=0.9.2-1&stamp=1534369425&raw=0).

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -39,6 +39,28 @@ jobs:
       - checkout
       - run: python setup.py egg_info
 
+  32bit:
+    docker:
+      - image: quay.io/pypa/manylinux1_i686
+    steps:
+      - checkout
+      - run: *skip-check
+      - run:
+          name: yum dependencies
+          command: yum install -y freetype freetype-devel libpng-devel tk
+      - run:
+          name: Install dependencies one
+          command: /opt/python/cp36-cp36m/bin/pip install --progress-bar off numpy scipy astropy matplotlib==1.5.3
+      - run:
+          name: Install dependencies two
+          command: /opt/python/cp36-cp36m/bin/pip install --progress-bar off sqlalchemy scikit-image==0.13.1 glymur drms suds-jurko beautifulsoup4 requests python-dateutil pytest pytest-cov pytest-mock pytest-xdist mock hypothesis pytest-astropy pytest-rerunfailures
+      - run:
+          name: Run tests
+          command: PYTHONHASHSEED=42 /opt/python/cp36-cp36m/bin/python setup.py test --parallel=4
+    environment:
+          MPLBACKEND: agg
+          COLUMNS: 180
+
   figure-tests-36:
     docker:
       - image: continuumio/miniconda3
@@ -85,6 +107,10 @@ workflows:
     jobs:
       - egg-info-36
       - egg-info-37
+
+  build-32:
+    jobs:
+      - 32bit
 
   figure-tests:
     jobs:

--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -11,7 +11,6 @@ include sunpy/data/sunpyrc
 include sunpy/tests/figure_hashes_*.json
 include setup.py
 include ah_bootstrap.py
-include ez_setup.py
 
 recursive-include astropy_helpers *
 exclude astropy_helpers/.*

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -1,0 +1,27 @@
+variables:
+  Agent.Source.Git.ShallowFetchDepth: 1
+
+jobs:
+- template: azure-templates.yml
+  parameters:
+    name: Linux_36_offline
+    os: linux
+    tox: py36-offline
+
+- template: azure-templates.yml
+  parameters:
+    name: macOS_37_offline
+    os: macos
+    tox: py37-offline
+
+- template: azure-templates.yml
+  parameters:
+    name: Windows_36_offline
+    os: windows
+    tox: py36-offline
+
+- template: azure-templates.yml
+  parameters:
+    name: Linux_37_online
+    os: linux
+    tox: py37-online -- -n=8

--- a/azure-templates.yml
+++ b/azure-templates.yml
@@ -1,0 +1,35 @@
+jobs:
+- job: ${{ format(parameters.name) }}
+  pool:
+    ${{ if eq(parameters.os, 'windows') }}:
+      vmImage: vs2017-win2016
+    ${{ if eq(parameters.os, 'macos') }}:
+      vmImage: macOS 10.13
+    ${{ if eq(parameters.os, 'linux') }}:
+      vmImage: Ubuntu 16.04
+
+  steps:
+  - task: CondaEnvironment@1
+    displayName: setup conda
+
+  - task: UsePythonVersion@0
+    displayName: setup python
+    inputs:
+      versionSpec: 3.7
+
+  - script: python -m pip install --upgrade pip setuptools
+    displayName: upgrade pip and setuptools
+
+  - script: pip install --upgrade tox
+    displayName: install tox
+
+  - script: pip install --upgrade git+https://github.com/drdavella/tox-conda
+    displayName: install tox-conda
+
+  - script: tox -vv -e ${{ parameters.tox }}
+    displayName: run tox-conda
+
+  - script: |
+      pip install --upgrade codecov
+      codecov
+    displayName: run codecov

--- a/changelog/2844.bugfix.rst
+++ b/changelog/2844.bugfix.rst
@@ -1,0 +1,1 @@
+Timeseries and lightcurve will now respect updated config values for download directory.

--- a/docs/guide/units-coordinates.rst
+++ b/docs/guide/units-coordinates.rst
@@ -165,7 +165,7 @@ frames defined in Astropy, for example `~astropy.coordimates.ICRS`.
 
   >>> c.transform_to('icrs')
   <SkyCoord (ICRS): (ra, dec, distance) in (deg, deg, km)
-    (49.85053118, 0.05723938, 1417577.02975451)>
+    (49.85053118, 0.05723938, 1417577.0297545...)>
 
 
 

--- a/setup.py
+++ b/setup.py
@@ -30,8 +30,6 @@ AUTHOR = metadata.get('author', 'The SunPy Community')
 AUTHOR_EMAIL = metadata.get('author_email', '')
 LICENSE = metadata.get('license', 'unknown')
 URL = metadata.get('url', 'https://sunpy.org')
-with open(os.path.join(os.path.abspath(os.path.dirname(__file__)), 'README.rst'), encoding='utf-8') as f:
-    LONG_DESCRIPTION = f.read()
 __minimum_python_version__ = metadata.get("minimum_python_version", "3.6")
 
 # Enforce Python version check - this is the same check as in __init__.py but
@@ -39,6 +37,9 @@ __minimum_python_version__ = metadata.get("minimum_python_version", "3.6")
 if sys.version_info < tuple((int(val) for val in __minimum_python_version__.split('.'))):
     sys.stderr.write("ERROR: SunPy requires Python {} or later\n".format(__minimum_python_version__))
     sys.exit(1)
+
+with open(os.path.join(os.path.abspath(os.path.dirname(__file__)), 'README.rst'), encoding='utf-8') as f:
+    LONG_DESCRIPTION = f.read()
 
 # Import ah_bootstrap after the python version validation
 import ah_bootstrap  # noqa

--- a/sunpy/conftest.py
+++ b/sunpy/conftest.py
@@ -38,7 +38,7 @@ def pytest_runtest_setup(item):
     pytest hook to skip all tests that have the mark 'remotedata' if the
     pytest_remotedata plugin is not installed.
     """
-    if isinstance(item, item.Function):
+    if isinstance(item, pytest.Function):
         if 'remote_data' in item.keywords and not HAVE_REMOTEDATA:
             pytest.skip("skipping remotedata tests as pytest-remotedata is not installed")
 

--- a/sunpy/coordinates/tests/test_ephemeris.py
+++ b/sunpy/coordinates/tests/test_ephemeris.py
@@ -16,7 +16,8 @@ def test_get_body_barycentric():
     assert_quantity_allclose(e1.radius, 0.9832947*u.AU, atol=5e-7*u.AU)
 
     e2 = get_body_heliographic_stonyhurst('earth', '2013-Sep-01')
-    assert_quantity_allclose(e2.lon, 0*u.deg, atol=1e-12*u.deg)
+    # https://github.com/sunpy/sunpy/issues/2727
+    assert_quantity_allclose((e2.lon+1*u.deg)%(360*u.deg), 1*u.deg, atol=1e-12*u.deg)
     assert_quantity_allclose(e2.lat, 7.19*u.deg, atol=5e-3*u.deg)
     assert_quantity_allclose(e2.radius, 1.0092561*u.AU, atol=5e-7*u.AU)
 

--- a/sunpy/util/config.py
+++ b/sunpy/util/config.py
@@ -46,22 +46,20 @@ def get_and_create_download_dir():
     '''
     Get the config of download directory and create one if not present.
     '''
-    config = load_config()
-    if not os.path.isdir(config.get('downloads', 'download_dir')):
-        os.makedirs(config.get('downloads', 'download_dir'))
+    if not os.path.isdir(sunpy.config.get('downloads', 'download_dir')):
+        os.makedirs(sunpy.config.get('downloads', 'download_dir'))
 
-    return config.get('downloads', 'download_dir')
+    return sunpy.config.get('downloads', 'download_dir')
 
 
 def get_and_create_sample_dir():
     '''
     Get the config of download directory and create one if not present.
     '''
-    config = load_config()
-    if not os.path.isdir(config.get('downloads', 'sample_dir')):
-        os.makedirs(config.get('downloads', 'sample_dir'))
+    if not os.path.isdir(sunpy.config.get('downloads', 'sample_dir')):
+        os.makedirs(sunpy.config.get('downloads', 'sample_dir'))
 
-    return config.get('downloads', 'sample_dir')
+    return sunpy.config.get('downloads', 'sample_dir')
 
 
 def print_config():

--- a/sunpy/util/config.py
+++ b/sunpy/util/config.py
@@ -184,7 +184,7 @@ def _expand_filepath(filepath, working_dir=""):
         return tempfile.gettempdir()
     # Relative filepaths
     elif not filepath.startswith("/"):
-        return os.path.join(working_dir, filepath)
+        return  os.path.abspath(os.path.join(working_dir, filepath))
     # Absolute filepath
     else:
-        return filepath
+        return os.path.abspath(filepath)

--- a/sunpy/util/tests/test_config.py
+++ b/sunpy/util/tests/test_config.py
@@ -1,0 +1,28 @@
+import os
+
+from sunpy import config
+from sunpy.util.config import get_and_create_sample_dir, get_and_create_download_dir
+
+USER = os.path.expanduser('~/')
+
+
+def test_get_and_create_download_dir():
+    # test default config
+    path = get_and_create_download_dir()
+    assert path == os.path.join(USER, 'sunpy', 'data')
+    # test updated config
+    new_path = os.path.join(USER, 'data_here_please')
+    config.set('downloads', 'download_dir', new_path)
+    path = get_and_create_download_dir()
+    assert path == os.path.join(USER, new_path)
+
+
+def test_get_and_create_sample_dir():
+    # test default config
+    path = get_and_create_sample_dir()
+    assert path == os.path.join(USER, 'sunpy', 'data', 'sample_data')
+    # test updated config
+    new_path = os.path.join(USER, 'data_here_please')
+    config.set('downloads', 'sample_dir', new_path)
+    path = get_and_create_sample_dir()
+    assert path == new_path

--- a/sunpy/util/tests/test_config.py
+++ b/sunpy/util/tests/test_config.py
@@ -3,7 +3,7 @@ import os
 from sunpy import config
 from sunpy.util.config import get_and_create_sample_dir, get_and_create_download_dir
 
-USER = os.path.expanduser('~/')
+USER = os.path.expanduser('~')
 
 
 def test_get_and_create_download_dir():

--- a/sunpy/util/tests/test_sysinfo.py
+++ b/sunpy/util/tests/test_sysinfo.py
@@ -2,6 +2,7 @@
 
 import sunpy
 
+
 def test_sysinfo():
 
     output = sunpy.util.get_sys_dict()

--- a/sunpy/util/tests/test_util.py
+++ b/sunpy/util/tests/test_util.py
@@ -3,9 +3,7 @@ This module tests the functions implemented in sunpy.util.util.
 """
 
 import numpy as np
-
 import pytest
-from pytest_mock import mocker
 
 from sunpy.util import util
 

--- a/tox.ini
+++ b/tox.ini
@@ -32,8 +32,7 @@ conda_deps =
     mock
     openjpeg
     pandas
-    # breaks too much
-    pytest<3.10
+    pytest
     pytest-astropy
     pytest-cov
     pytest-mock

